install/build.sh

```diff
@@ -0,0 +1,76 @@
+#! /usr/bin/env sh
+
+set -e
+
+GITHUB_SHA=${GITHUB_SHA:-$(git rev-parse HEAD)}
+GITHUB_REF=${GITHUB_REF:-$(git rev-parse --abbrev-ref HEAD)}
+GITHUB_REF=${GITHUB_REF##*/}
+GITHUB_REPOSITORY=${GITHUB_REPOSITORY:-"starship"}
+
+IMAGE="debian:latest"
+PLATFORMS=
+
+while :; do
+		case $1 in
+			-p|--platform)
+				if [ -z "${PLATFORMS:-}" ]; then
+					PLATFORMS=$2
+				else
+					PLATFORMS="$2,$PLATFORMS"
+				fi
+				shift
+				;;
+			-i|--image)
+				IMAGE=$2
+				shift
+				;;
+			--debug)
+				set -x
+				;;
+			?*)
+				printf "\n%s\n" "error: unknown argument: $1"
+				exit 1
+				;;
+			*)
+				break
+				;;
+		esac
+		shift
+	done
+
+
+if [ -z "${PLATFORMS:-}" ]; then
+	PLATFORMS="linux/arm64,linux/amd64"
+fi
+
+if [ -n "${GITHUB_TOKEN:-}" ]; then
+	printf "%s" "$GITHUB_TOKEN" | docker login ghcr.io -u github --password-stdin 1>/dev/null 2>&1
+fi
+
+IMAGE_NAME=${IMAGE%%:*}
+IMAGE_VERSION=${IMAGE##*:}
+
+TAG="${IMAGE_NAME}-${IMAGE_VERSION}-${GITHUB_REF}"
+
+printf "\nBUILDING:\n  PLATFORMS : %s\n  IMAGE     : %s\n  TAG       : %s\n\n" \
+	"$PLATFORMS" "$IMAGE" "$TAG"
+
+# create the builder
+docker buildx create --name starship --node starship0 --driver docker-container --use
+
+# build the image
+docker buildx build \
+    --progress=plain \
+    --output=type=oci,dest=starship-"${TAG}".tar.gz \
+	--platform="$PLATFORMS" \
+	--build-arg IMAGE="$IMAGE" \
+	--label=org.opencontainers.image.name=starship \
+	--label=org.opencontainers.image.revision="$GITHUB_SHA" \
+	--label=org.opencontainers.image.version="$GITHUB_REF" \
+	--label=org.opencontainers.image.source="https://github.com/$GITHUB_REPOSITORY/starship" \
+	--tag "ghcr.io/$GITHUB_REPOSITORY/starship:$TAG" \
+	"${PWD}/install"
+
+    # this can be added if images are ever published
+	# --cache-from=type=registry,ref="ghcr.io/$GITHUB_REPOSITORY/starship:$IMAGE-latest" \
```

Comment:
Is there a use-case to distribute Starship in a container?

Reply:
Not that I can think of except to be able to quickly pull one to reproduce an issue someone reports with a specific flavour combination. I can't think of any reason beyond that.